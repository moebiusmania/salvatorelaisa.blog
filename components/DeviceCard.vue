<script setup lang="ts">
import type { ParsedContent } from "@nuxt/content/dist/runtime/types";
import External from "./icons/External.server.vue";
import Doc from "./icons/Doc.server.vue";

defineProps<{
  device: ParsedContent;
}>();

const formatDate = (date: string): string => {
  const parts = date.split("-");
  return parts.length === 1 ? date : new Date(date).toLocaleDateString("it");
};
</script>

<template>
  <article class="bg-neutral-content">
    <figure>
      <img :src="device.image" :alt="device.title" />
    </figure>
    <div class="dark:text-base-100">
      <div>
        <span v-for="tag in device.tags" class="sl-badge">{{ tag }}</span>
      </div>
      <p>
        Data di acquisto:
        {{ formatDate(device.purchase) }}
      </p>
      <h2>
        <NuxtLink class="hover:text-primary" :href="device.post || device.url">
          {{ device.title }}
        </NuxtLink>
        <External v-if="device.url" />
        <Doc v-if="device.post" />
      </h2>
      <ContentRenderer :value="device" />
    </div>
  </article>
</template>

<style>
@import url('./Badge.css');
</style>

<style scoped>
/* 
Card component from DaisyUI
https://daisyui.com/components/card/
*/

article {
  border-style: none;
  box-shadow: 0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04);
  position: relative;
  display: flex;
  flex-direction: column;
  border-radius: 4px;

  & figure {
    padding: 1rem;
    background-color: #ffffff;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    border-start-start-radius: inherit;
    border-start-end-radius: inherit;
    border-end-start-radius: unset;
    border-end-end-radius: unset;
  }

  &>div {
    display: flex;
    flex: 1 1 auto;
    flex-direction: column;
    gap: 0.5rem;
    padding: 32px;

    & span {
      margin-right: 0.5rem;
      text-decoration: none;
    }

    &>p {
      flex: none;
      font-size: 0.875rem;
      line-height: 1.25rem;
    }

    &>h2 {
      display: flex;
      align-items: center;
      gap: 0.5rem;
      font-size: 1.25rem;
      line-height: 1.75rem;
      font-weight: 600;

      & a {
        text-decoration: underline;
      }

      & svg {
        display: inline-block;
        width: 1rem;
        height: 1rem;
      }
    }
  }
}
</style>
