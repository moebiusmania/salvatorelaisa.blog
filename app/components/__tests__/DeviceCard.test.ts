import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import DeviceCard from "../DeviceCard.server.vue";

describe("DeviceCard", () => {
  it("matches snapshot", () => {
    const mockDevice = {
      title: "Test Device",
      image: "/test-image.jpg",
      tags: ["tag1", "tag2"],
      purchase: "2024-03-20",
      url: "https://example.com",
      body: "Test device description",
    };

    const wrapper = mount(DeviceCard, {
      props: {
        device: {
          ...mockDevice,
          _id: "test-id",
          body: {
            type: "root",
            children: [{ type: "text", value: "Test device description" }],
          },
        },
      },
      global: {
        components: {
          // Mock the Nuxt-specific components
          NuxtLink: {
            template: '<a href=""><slot /></a>',
            props: ["href"],
          },
          ContentRenderer: {
            template: "<div><slot /></div>",
            props: ["value"],
          },
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
