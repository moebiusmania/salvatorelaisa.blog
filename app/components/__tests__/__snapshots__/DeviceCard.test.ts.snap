// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeviceCard > matches snapshot 1`] = `
"<article data-v-26916834="">
  <figure data-v-26916834=""><img data-v-26916834="" src="/test-image.jpg" alt="Test Device"></figure>
  <div data-v-26916834="">
    <div data-v-26916834=""><span data-v-26916834="" class="sl-badge">tag1</span><span data-v-26916834="" class="sl-badge">tag2</span></div>
    <p data-v-26916834=""> Data di acquisto: 20/03/2024</p>
    <h2 data-v-26916834=""><a data-v-26916834="" href="https://example.com" rel="noopener noreferrer">Test Device</a>
      <external data-v-26916834=""></external>
      <!--v-if-->
    </h2>
    <!---->
  </div>
</article>"
`;
